### PR TITLE
[BEAM-8815] Define the no artifacts retrieval token in proto

### PR DESCRIPTION
**Pull request by tweise**
Is there a good way to avoid repetition of `extend google.protobuf.EnumValueOptions` for this purpose?

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: model/job-management/src/main/proto/beam_artifact_api.proto [hunk 1]
```diff
@@ -147,3 +154,9 @@ message CommitManifestResponse {
   // ArtifactRetrievalService.
   string retrieval_token = 1;
 }
+
+// repeated from beam_runner_api.proto - is there a way to share the extension?
```
Comment (ibzib) [inline model/job-management/src/main/proto/beam_artifact_api.proto]: Looks like you can do it if you include the package name: https://stackoverflow.com/questions/17126230/extending-protobuf-fieldoptions-in-imported-proto-file

File: model/job-management/src/main/proto/beam_artifact_api.proto [hunk 2]
```diff
@@ -147,3 +154,9 @@ message CommitManifestResponse {
   // ArtifactRetrievalService.
   string retrieval_token = 1;
 }
+
+// repeated from beam_runner_api.proto - is there a way to avoid this
```
[the remark this answers is not in the document]
Comment (ibzib) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: Whoops, ignore this and see my other comment

File: model/job-management/src/main/proto/beam_artifact_api.proto (hunk 1 above)
Comment (tweise) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: Following doesn't make the extension available, even though the import is OK: `import "beam_runner_api.proto"`

Comment (ibzib) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: What error are you getting?

It looks like the import needs to be fully qualified, i.e. `org.apache.beam.model.pipeline.v1.beam_constant` 
https://github.com/apache/beam/blob/2b441c25c6353f1f293474c144046e241a0f2a02/model/pipeline/src/main/proto/metrics.proto#L221

Comment (tweise) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: That was it, thanks!

File: model/job-management/src/main/proto/beam_artifact_api.proto [hunk 3]
```diff
@@ -67,6 +69,10 @@ message ArtifactMetadata {
 
 // A collection of artifacts.
 message Manifest {
+  enum Constants {
```
Comment (ibzib) [inline model/job-management/src/main/proto/beam_artifact_api.proto]: I don't think `Manifest` is the right place for this.

The protobuf spec seems to support free-standing enums, but it looks like the constant enums elsewhere in Beam are wrapped in messages, causing redundant naming: `Api.BeamConstants.Constants.CONSTANT`. However, I'm not sure if that was deliberate, and if so what the reason might be.

Comment (tweise) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: The constant is wrapped in the message (`Manifest`) and the reference is `ArtifactApi.Manifest.Constants.NO_ARTIFACTS_STAGED_TOKEN` https://github.com/apache/beam/pull/10285/files#diff-aed248327fcc4e847968c709c2fa5a1aR56

Comment (ibzib) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: I don't like that while `Constants` is scoped to `Manifest`, `Manifest` doesn't use `Constants`. Though admittedly this is a bit nit-picky.

Comment (lukecwik) [reply model/job-management/src/main/proto/beam_artifact_api.proto]: The proto code generator will place enums into the root namespace and not use the name of the enum as a namespace for some languages while in other languages like Java the enum's name is part of the namespace which is why they are wrapped inside a message.

We ran into this issue with the enums which used the same constant `UNSPECIFIED` where the code generation had issues.
See

**Comment by tweise** [comment]
Run Java PreCommit

**Comment by tweise** [comment]
Run Java PreCommit

**Comment by tweise** [comment]
Run Python PreCommit

**Comment by tweise** [comment]
Run Java PreCommit

**Comment by tweise** [comment]
Run Java PreCommit

**Comment by tweise** [comment]
Run Python PreCommit

**Pull request merged**